eventmesh-runtime/src/main/java/com/webank/eventmesh/runtime/core/protocol/http/processor/HeartBeatProcessor.java

```diff
@@ -145,7 +145,26 @@ public void processRequest(ChannelHandlerContext ctx, AsyncContext<HttpCommand>
         }
         synchronized (proxyHTTPServer.localClientInfoMapping){
             for (Map.Entry<String, List<Client>> groupTopicClientMapping : tmp.entrySet()) {
-                proxyHTTPServer.localClientInfoMapping.put(groupTopicClientMapping.getKey(), groupTopicClientMapping.getValue());
+                List<Client> localClientList =  proxyHTTPServer.localClientInfoMapping.get(groupTopicClientMapping.getKey());
+                if (CollectionUtils.isEmpty(localClientList)){
+                    proxyHTTPServer.localClientInfoMapping.put(groupTopicClientMapping.getKey(), groupTopicClientMapping.getValue());
+                }else {
+                    List<Client> tmpClientList = groupTopicClientMapping.getValue();
+                    for (Client tmpClient : tmpClientList){
```

Comment:
```suggestion
                    for (Client tmpClient : tmpClientList){
                        boolean isContains = false;
                        for (Client localClient : localClientList){
                            if (StringUtils.equals(localClient.url, tmpClient.url)){
                                isContains = true;
                                break;
                            }
                        }
                        if (!isContains){
                            localClientList.add(tmpClient);
                        }
                    }
```
These code can be simplified and extracted as a method will be better

Reply:
ok, I will fix that.